modules/contact/test/tests/bouncing-block-contact_pdass/try_gitterle.i

```diff
@@ -0,0 +1,274 @@
+  []
+  [vel_variable]
+    block = 3
+    use_dual = false
+  []
+  [tangential_lm]
+    block = 3
+    use_dual = true
+  []
+[]
+
+[ICs]
+  [disp_y]
+    block = 2
+    variable = disp_y
+    value = '${fparse starting_point + offset}'
+    type = ConstantIC
+  []
+[]
+
+[Modules/TensorMechanics/Master]
+  [all]
+    strain = FINITE
+    generate_output = 'stress_xx stress_yy'
+    block = '1 2'
+  []
+[]
+
+[Materials]
+  [elasticity_2]
+    type = ComputeIsotropicElasticityTensor
+    block = '2'
+    youngs_modulus = 1e3
+    poissons_ratio = 0.3
+  []
+  [elasticity_1]
+    type = ComputeIsotropicElasticityTensor
+    block = '1'
+    youngs_modulus = 1e6
+    poissons_ratio = 0.3
+  []
+  [stress]
+    type = ComputeFiniteStrainElasticStress
+    block = '1 2'
+  []
+[]
+
+[Constraints]
+  [weighted_gap_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+  []
+
+ # Create gap_variable
+  [weighted_gap_indicator]
+    type = ComputeWeightedGapLMMechanicalContactIndicator
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = gap_variable
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+  []
+
+ # Create vel_variable
+  [tangential_vel]
+    type = ComputeWeightedTangentialVelocityLMMechanicalContactIndicator
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = vel_variable
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+    contact_pressure = normal_lm
+  []
+
+  [ncp_lm]
+    type = ApplyPenetrationConstraintLMMechanicalContact
+    secondary = 10
+    primary = 20
+    variable = normal_lm
+    primary_variable = disp_x
+    c = 1.0e4
+  []
+
+  [normal_x]
+    type = NormalMortarMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [normal_y]
+    type = NormalMortarMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [friction]
+    type = ApplyGitterleFrictionConstraintLMMechanicalContact
+    primary = 20
+    secondary = 10
+    variable = tangential_lm
+    primary_variable = disp_x
+    contact_pressure = normal_lm
+    gap_value = gap_variable
+    weighted_tangential_velocity = vel_variable
+    disp_y = disp_y
+    c_n = 1.0e3
+    c_t = 1.0e5
+    mu = 0.12
+  []
```

Comment:
This is the object that computes PDASS tangential constraints proposed by Wohlmuth.